lis/surfacemodels/land/noahmp.4.0.1/da_soilm_sig0VVVH/noahmp401_getSig0VVVHpred.F90

```diff
@@ -0,0 +1,127 @@
+!-----------------------BEGIN NOTICE -- DO NOT EDIT-----------------------
+! NASA Goddard Space Flight Center
+! Land Information System Framework (LISF)
+! Version 7.3
+!
+! Copyright (c) 2020 United States Government as represented by the
+! Administrator of the National Aeronautics and Space Administration.
+! All Rights Reserved.
+!-------------------------END NOTICE -- DO NOT EDIT-----------------------
+!BOP
+! !ROUTINE: noahmp401_getSig0VVVHpred
+! \label{noahmp401_getSig0VVVHpred}
+!
+! !REVISION HISTORY:
+! 15 Mar 2021: Sara Modanesi; Initiated specifications
+!
+! !INTERFACE:
+
+subroutine noahmp401_getSig0VVVHpred(n,k,obs_pred)
+
+! !USES:
+
+  use ESMF
+  !use LIS_coreMod, only : LIS_rc,LIS_surface
+  use LIS_RTMMod,  only : LIS_forwardState, LIS_RTM_run
+  !use LIS_logMod,  only : LIS_verify
+
+  use noahmp401_lsmMod
+
+  ! added
+  use LIS_constantsMod
+  use LIS_coreMod
+  use LIS_logMod
+  use LIS_dataAssimMod
+  use LIS_DAobservationsMod
+
+!EOP
+
+  implicit none
+! !ARGUMENTS: 
+  integer, intent(in)    :: n
+  integer, intent(in)    :: k
+  real                   :: obs_pred(LIS_rc%obs_ngrid(k)*2,LIS_rc%nensem(n))
+  real                   :: obs_pred_VV(LIS_rc%obs_ngrid(k),LIS_rc%nensem(n))
+  real                   :: obs_pred_VH(LIS_rc%obs_ngrid(k),LIS_rc%nensem(n))
+  integer                :: count1(LIS_rc%obs_ngrid(k)*2,LIS_rc%nensem(n))
+!
+! !DESCRIPTION:
+!
+!  Returns the Backscatter VV and VH obs pred (model's estimate of 
+!  observations) for data assimilation
+! 
+!  The arguments are: 
+!  \begin{description}
+!  \item[n] index of the nest \newline
+!  \item[obs\_pred] model's estimate of observations \newline
+!  \end{description}
+!EOP
+
+  integer                :: i,t,m,gid
+  real, pointer          :: s0VV(:), s0VH(:)
+  type(ESMF_Field)       :: varField
+  integer                :: status
+
+!!call the forward model
+call LIS_RTM_run(n)
+
+  call ESMF_StateGet(LIS_forwardState(n), "WCM_Sig0VV", varField, rc=status)
+!LIS_histDAtaMod
+  call LIS_verify(status, &
+       "Error in StateGet in noahmp401_getSig0Pred for WCM_Sig0VV")
+
+  call ESMF_FieldGet(varField, localDE=0,farrayPtr=s0VV, rc=status)!variable
+!s0VV previously defined
+  call LIS_verify(status, &
+       'Error in FieldGet in noahmp401_getSig0Pred for WCM_Sig0VV')
+
+  call ESMF_StateGet(LIS_forwardState(n), "WCM_Sig0VH", varField, rc=status)
+  call LIS_verify(status, &
+       "Error in StateGet in noahmp401_getSig0Pred for WCM_Sig0VH")
+
+  call ESMF_FieldGet(varField, localDE=0,farrayPtr=s0VH, rc=status)
+  call LIS_verify(status, &
+       'Error in FieldGet in noahmp401_getSig0Pred for WCM_Sig0VH')
+  obs_pred = 0.0
+  count1 = 0.0
+  do i=1,LIS_rc%npatch(n,LIS_rc%lsm_index),LIS_rc%nensem(n)
+     do m=1,LIS_rc%nensem(n)
+        t = i+m-1
+        gid = LIS_surface(n,LIS_rc%lsm_index)%tile(t)%index
+        obs_pred(gid,m)= obs_pred(gid,m) + s0VV(t)
+        count1(gid,m) = count1(gid,m) +1
+     enddo
+  enddo
+
```

Comment:
(1) count1 for s0VV is never used, i.e. no average s0VV is computed? (2) The order of routines for VV and VH is different and not intuitive. (3) what is done with obs_pred_VV/VH?

Reply:
(1) count1 is used for s0VV and s0VH simultaneously at end of the subroutine. I added a comment for clarification. 
(2) correct, thanks for pointing this out. It is changed now. I also outcommented an unused section to keep it for possible later use.
(3) nothing, see (2)

Reply:
looks better now!